### ISSUE TITLE
Push to branch without a new commit doesn't trigger workflow

### ISSUE THREAD

**Issue by tomaskraina** [bitrise-io/bitrise.io#13]
When pushing commits that already exist in one branch (e.g. `develop`) to another branch (e.g. `master`) without merge or adding a new commit, this push doesn't trigger a workflow in bitrise. Not even a workflow for `*` wildcard. I've checked githubs webhooks and the push even is posted correctly.

In contrast, when a branch (`develop`) is merged to another one (`master`) on github, it triggers the particular workflow.

**Comment by birmacher** [comment]
Hi @tomaskraina!

That's because when we get a webhook from GitHub we check the `distinct` value inside the json, and only build those commits that has `distinct` set to `true`. That means that the commit was already pushed to another branch or not.

So if you are pushing a code that already was tested it don't make sense to me to re-run the tests again.

I'm adding @viktorbenei so you can check if we should move forward with it or stay as it is right now.

**Comment by viktorbenei** [comment]
Do you have a use case @tomaskraina where triggering a new build in a case like this would be useful? We're curious, maybe we missed something and we should make this behavior optional, similar to Pull Request testing.

**Comment by tomaskraina** [comment]
@viktorbenei yes, would like to use it to trigger a workflow that creates a new build.

For example:
a push to `release/crashlytics` would trigger a workflow that makes a build and uploads it to crashlytics beta.

**Comment by viktorbenei** [comment]
I'll add this to our list @tomaskraina , thank you for the feedback! That said you can do this with the current system as well, if you `merge` between branches instead of `rebasing`, as merge generates a new commit which triggers a new build.

**Comment by tomaskraina** [comment]
ok, thanks @viktorbenei

**Comment by kylejm** [comment]
+1 for this! :+1: 

We have a similar thing with merging into release branches etc.

**Comment by viktorbenei** [comment]
It's on our list, but not ETA yet.

**Comment by viktorbenei** [comment]
Broadly, it's on the list of "for early next year" ;)

**Comment by rothmichaels** [comment]
> Broadly, it's on the list of "for early next year" ;)
:+1: Looking forward to it ;)

**Comment by viktorbenei** [comment]
We're working on a new, open source webhook processor which will now trigger builds for non `distinct` commits too. We'll let you know when it's available, if you want to beta test it - ETA is end of this week.

**Comment by viktorbenei** [comment]
The new GitHub and Bitbucket V2 ("Webhooks") implementation are now available as beta on https://hooks.bitrise.io - repository: https://github.com/bitrise-io/bitrise-webhooks

This new processor now triggers builds for non distinct commits/pushes. Feel free to contact us if you'd like to test it.

**Comment by viktorbenei** [comment]
The new https://hooks.bitrise.io service is now publicly available, and integrated into bitrise.io . If you use the GitHub / Bitbucket automatic webhook registration it'll register a https://hooks.bitrise.io hook URL for your project, and the `Code` tab was also updated for the new hooks service.

This new service is completely open source, and it now doesn't reject non `distinct` code pushes, those will trigger builds too.

We'll soon publish a blog post too, but hopefully you have every info on the `Code` tab to replace/update your existing webhooks with the new https://hooks.bitrise.io service hook URL. If you have any question feel free to ask.

I'll close this issue now, as I believe we covered everything what was requested here - thank you for the suggestion and for all the information, it really helped a lot!
